### PR TITLE
Refactor Team Stats

### PR DESCRIPTION
**Pull request by smcunning**
PR Notes - 
- Refactored methods for all team stats
- Added tests for all methods and helper methods 
- Updated variable and method names for clarity 
- All tests pass
- Passes spec harness

### CODE REVIEW

File: lib/game_teams_manager.rb [hunk 1]
```diff
@@ -59,4 +59,117 @@ def add_wins_losses(gt_results, coach_record_start)
...
+
+  def team_games_by_season(team_id)
+    team_games_by_season = {}
+    game_info_by_team(team_id).each do |game|
+      (team_games_by_season[game.game_id.to_s[0..3]] ||= []) << game
+    end
+    team_games_by_season
+  end
+
+  def win_percentage_by_season(team_id)
```
Comment (smcunning) [inline lib/game_teams_manager.rb:194]: This method is over 10 lines - would love to refactor this to be more succinct!

File: lib/game_teams_manager.rb [hunk 2]
```diff
@@ -59,4 +59,117 @@ def add_wins_losses(gt_results, coach_record_start)
...
+    end.goals.to_i
+  end
+
+  def fewest_goals_scored(team_id)
+    game_info_by_team(team_id).min_by do |game|
+      (game.goals).to_i
+    end.goals.to_i
+  end
+
+  def favorite_opponent(team_id)
```
Comment (smcunning) [inline lib/game_teams_manager.rb:165]: This method used to be 12 lines long! 💯

File: lib/game_teams_manager.rb [hunk 3]
```diff
@@ -59,4 +59,117 @@ def add_wins_losses(gt_results, coach_record_start)
...
   end
+
+#-------------TeamStats
+  def best_season(team_id)
+    best_season = win_percentage_by_season(team_id).max_by do |season, wins_percent|
+        wins_percent
+      end
+      best_year = best_season[0].to_i
+      "#{best_year}201#{best_year.digits[0] + 1}"
+    end
```
Comment (taphill) [inline lib/game_teams_manager.rb]: The spacing on these `ends` looks like it is a space off

File: lib/game_teams_manager.rb (hunk 2 above)
Comment (taphill) [reply lib/game_teams_manager.rb:165]: Good job!!!

File: lib/game_teams_manager.rb (hunk 1 above)
Comment (taphill) [reply lib/game_teams_manager.rb:194]: We can talk bout it during refactoring, but I think it's fine for now!

**Comment by taphill** [comment]
Good job on this @smcunning!

File: lib/game_teams_manager.rb (hunk 3 above)
Comment (smcunning) [reply lib/game_teams_manager.rb]: Updated, thanks for finding that!

**Pull request merged**